src/bundles/ApplicationsAdmin/components/Agency/Agency.js

```diff
@@ -0,0 +1,194 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { AUcheckbox } from '@gov.au/control-input'
+import AUselect from '@gov.au/select'
+import AUtextInput from '@gov.au/text-inputs'
+import { saveAgency } from '../../redux/modules/agency'
+import format from 'date-fns/format'
+
+class Agency extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      loading: false,
+      agency: null,
+      saved: null
+    }
+
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  componentDidMount() {
+    this.setState({
+      agency: this.props.agency
+    })
+  }
+
+  handleSubmit(event) {
+    event.preventDefault()
+    const formData = new FormData(event.target)
+    let data = Object.fromEntries(formData)
+    const { agency } = this.state
+    data.id = agency.id
+    if (data.reports === "on") {
+      data.reports = true
+    } else {
+      data.reports = false
+    }
+    if (data.whitelisted === "on") {
+      data.whitelisted = true
+    } else {
+      data.whitelisted = false
+    }
+     data.domains = data.domains.split('\n')
+
+    this.setState({
+      loading: true
+    })
+    this.props.saveAgency(data)
+    .then(r => {
+      if (r.status === 200) {
+        return r.json()
+      }
+      this.setState({
+        loading: false
+      })
+      return Promise.reject()
+    })
+    .then(r => {
+      this.setState({
+        loading: false,
+        agency: r,
+        saved: new Date()
```

Comment:
These state changes can probably occur inside the `r.status === 200` condition in the `.then()` above and this `.then()` can be removed, as right now I think these will be called if the Promise is rejected.

Reply:
updated